service/frontend/workflowHandler.go

```diff
@@ -3011,37 +3018,594 @@ func (wh *WorkflowHandler) ListTaskQueuePartitions(ctx context.Context, request
+	}
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, err
+	}
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = wh.historyClient.TerminateWorkflowExecution(ctx, &historyservice.TerminateWorkflowExecutionRequest{
+		NamespaceId: namespaceID.String(),
+		TerminateRequest: &workflowservice.TerminateWorkflowExecutionRequest{
+			Namespace:         request.Namespace,
+			WorkflowExecution: &commonpb.WorkflowExecution{WorkflowId: request.ScheduleId},
+			Reason:            "terminated by DeleteSchedule",
+			Identity:          request.Identity,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &workflowservice.DeleteScheduleResponse{}, nil
 }
 
 // List all schedules in a namespace.
 func (wh *WorkflowHandler) ListSchedules(ctx context.Context, request *workflowservice.ListSchedulesRequest) (_ *workflowservice.ListSchedulesResponse, retError error) {
-	return nil, serviceerror.NewUnimplemented("todo")
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if wh.isStopped() {
+		return nil, errShuttingDown
+	}
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, err
+	}
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	if request.GetMaximumPageSize() <= 0 {
+		request.MaximumPageSize = int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
+	}
+
+	if wh.isListRequestPageSizeTooLarge(request.GetMaximumPageSize(), request.GetNamespace()) {
+		return nil, serviceerror.NewInvalidArgument(fmt.Sprintf(errPageSizeTooBigMessage, wh.config.ESIndexMaxResultWindow()))
+	}
+
+	namespaceName := namespace.Name(request.GetNamespace())
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: should we respect this here?
+	if wh.config.DisableListVisibilityByFilter(namespaceName.String()) {
+		return nil, errNoPermission
+	}
+
+	persistenceResp, err := wh.visibilityMrg.ListOpenWorkflowExecutionsByType(ctx, &manager.ListWorkflowExecutionsByTypeRequest{
+		ListWorkflowExecutionsRequest: &manager.ListWorkflowExecutionsRequest{
+			NamespaceID:   namespaceID,
+			Namespace:     namespaceName,
+			PageSize:      int(request.GetMaximumPageSize()),
+			NextPageToken: request.NextPageToken,
+		},
+		WorkflowTypeName: scheduler.WorkflowType,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	schedules := make([]*schedpb.ScheduleListEntry, len(persistenceResp.Executions))
+	for i, ex := range persistenceResp.Executions {
+		schedules[i] = &schedpb.ScheduleListEntry{
+			ScheduleId: ex.GetExecution().GetWorkflowId(),
+			Memo:       ex.GetMemo(),
+			// TODO: is this correct? where does search attribute mapping happen?
+			SearchAttributes: ex.GetSearchAttributes(),
```

Comment:
cc @alexshtin

Reply:
turns out it happens inside the ES visibility manager, and the other visibility stores don't support search attributes so they don't do it

Reply:
which is not a good example of responsibility separation and needs to be changed one day.